### DISCUSSION TITLE
Game controllers support

### DISCUSSION THREAD

**Discussion by faergeek** [redeclipse/base#1154] [Ideas and Feature Requests]
Are there any plans to add game controllers support? It would be awesome to be able to play this awesome game using gamepad.

At first it might seem like you can't play FPS using gamepad without crashing UX for keyboard/mouse users. But there are different controllers utilising accelerometers and other such things for aiming, e.g. Steam Controller. And at least I would buy something that looks like rifle or something like that just to play instagib with it (my favourite mode BTW), ideally with VR. I think it would be super fun 👍 

I can't code C/C++, but I would donate some money for that. Actually I'm going to donate anyway :-). Or you could use https://www.bountysource.com/ or something like that if you are not going to implement it yourself, but would like somebody else to implement it. I would fund that too.

Thanks for great game!!!

**Comment by qreeves** [comment]
There's always been some sort of half-hearted idea to add in controller support, but the cost to benefit ratio for this in an FPS isn't high enough. I will do commissioned integrations for RE, but this is on a case by case basis and depends on many factors. You should email contact@redeclipse.net with a full proposal, including what's expected and how much you're willing to contribute, etc.

**Comment by faergeek** [comment]
If I understand correctly you're expecting something more or less spec-like. Are there any examples of that? Sorry, I always ask for examples of such things, I just can't do that good enough by myself :-)

**Comment by faergeek** [comment]
Is it preferrable to save all mechanics on gamepad as on keyboard/mouse? E.g. I asked a friend, he proposed separate buttons for wall run and punching. Should I avoid something like that?

**Comment by faergeek** [comment]
Is there something wrong about what I written? If there's no example of spec, I would try to come up with something by myself. I just need an answer

**Comment by qreeves** [comment]
Sorry, I've just had a lot on my plate lately and haven't had the time to get to these issues. I'm not ignoring them, they're just still sitting in my todo pile.

I don't really require specs or anything, just a list of things you'd expect to get out of any paid work, if that's what you're after. I don't like to take money unless there's an agreement about the conditions of said payment, and I'm confident I can meet them.

That being said, I have to finish what I'm already being paid to do before I can take on any more commitments right now. I know for a fact that the Steam integration will offer much more in the way of controller support, so this is a positive anyway.

**Comment by faergeek** [comment]
Sorry, I didn't meant anything like "I pay you, do this shit". I donate anyway. Just suggest that if you don't want to do that, but would accept someone else's PR for that, it's the way to go (bountysource).

**Comment by qreeves** [comment]
Oh, right. I'll do it eventually when I finish the Steamworks integration, but I'm happy for PR's to be submitted so long as the submitter follows the general style of the code in RE (it can be a bit weird and we're actually rather strict about style).

**Comment by faergeek** [comment]
Some autoformatting tool would help I think. At least you'll get rid of simple mistakes about code style. E.g. https://clang.llvm.org/docs/ClangFormat.html
Actually, don't judge it if you can't tweak it to your preference. I resisted tools like this myself. But now I can't even think of formatting code by hands :-)
